packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -123,6 +123,21 @@ class UpdatePackagesCommand extends FlutterCommand {
         temporaryDirectory.deleteSync(recursive: true);
       }
 
+      // The transitive dependency tree for the fake package does not contain
+      // dependencies between Flutter SDK packages and pub packages. We add them
+      // here.
+      for (PubspecYaml pubspec in pubspecs) {
+        final String package = pubspec.name;
+        final String version = pubspec.version;
+        for (PubspecDependency dependency in pubspec.dependencies) {
+          if (dependency.kind == DependencyKind.normal) {
+            tree._versions[package] = version;
+            tree._dependencyTree[package] ??= new Set<String>();
+            tree._dependencyTree[package].add(dependency.name);
+          }
+        }
+      }
```

Comment:
is this too aggressive?

We only want to be adding the dependencies that we took out ourselves, so that's only the ones that match your new function.

Reply:
This is necessary because otherwise we won't get transitive dependencies _through_ SDK dependencies (something needs to provide a link from a SDK package to a pub package). I added a comment above explaining this.